src/hard_coded_unrolled_functions.jl

```diff
@@ -0,0 +1,112 @@
+@inline _unrolled_mapcall(::Val{0}, f, op, itr) = op()
```

Comment:
Organizationally, I kind of like the hard-coded versions to be next to the generalized versions, is the main part of the code too difficult to navigate that way?

Reply:
I've updated #25 and rebased this branch. Both versions are now in "manually_unrolled_functions.jl"